src/mapping.rs

```diff
@@ -0,0 +1,313 @@
+//! A Parser for Proguard Mapping Files.
+//!
+//! The mapping file format is described
+//! [here](https://www.guardsquare.com/en/products/proguard/manual/retrace).
+
+#[cfg(feature = "uuid")]
+use uuid::Uuid;
+
+/// A Proguard Mapping file.
+pub struct ProguardMapping<'s> {
+    source: &'s [u8],
+}
+
+impl<'s> ProguardMapping<'s> {
+    /// Create a new Proguard Mapping.
+    pub fn new(source: &'s [u8]) -> Self {
+        Self { source }
+    }
+
+    /// Calculates the UUID of the mapping file.
```

Comment:
_nit_: Add some more information on what this does. Otherwise, it might suggest that the uuid is an inherent property of proguard mapping files, which they are not.